lib/messages/create/create-deployment-dialog.js

```diff
@@ -37,10 +37,11 @@ module.exports = ({ repository, branches, tags }) => {
         }],
       },
       {
-        label: 'Branch or tag to deploy',
-        type: 'select',
+        type: 'text',
+        label: 'Branch/tag/sha to deploy',
```

Comment:
I think changing this from a select to a text control makes it less useful, it is of great benefit to just be able to select an existing branch or tag from a list.

Having to type refs in by hand will be much more error-prone.

Reply:
But a select drop-down does not scale with the number of potential branches and tags a repository may have. In our case, we tag every build, and because of the current integrations limitation of 100 select items, we are forced to delete older tagged releases (something I'd rather we didn't do). Free-text input is error-prone I agree, but I personally would prefer this trade-off.

Reply:
I wouldn't prefer this option though. I think the user experience is a lot worse, for the sake of a small number of users who have a lot of tags/branches. Is there a compromise? Can we support some combination of select and free-form text entry?

Reply:
I'm open to your suggestion. Please read the slack API docs and get back to us.

Reply:
How about the Slack external source select element, combined with the a GitHub ref search?

https://api.slack.com/reference/block-kit/block-elements#external_select
https://developer.github.com/v3/git/refs/#list-matching-references

I don't know the Slack or GitHub APIs very well, so what I suggest may not be possible, but at first glance both APIs support searches based on partial matches of text.

I think going from a select field to a free-form entry field would be a backward step, and I would encourage looking at all possible solutions. At the moment, non-developers can trigger deployments via Slack, as they do not have to have intimate knowledge of tags, branches etc. The select dialog helps them to choose by displaying known branches and tags.

Two other options I can think of in case the above is not workable:
* Having both the current select control, and the text box, with the one being populated taking precedence.
* Having the option to configure which type of control (select or text) to use in the Slack app settings.

Reply:
Please ignore my comments above - let's get this merged so people can use this integration. One can always iterate on improvements for selecting tags/branches later.